packages/intl/src/message.ts

```diff
@@ -103,38 +105,38 @@ function serializeAstTag(node: TagNode, result: { value: string }) {
       result.value += '**';
       serializeAst(node[AstNodeIndices.Children], result);
       result.value += '**';
-      return;
+      break;
     case '$i':
       result.value += '*';
       serializeAst(node[AstNodeIndices.Children], result);
       result.value += '*';
-      return;
+      break;
     case '$code':
       result.value += '`';
       serializeAst(node[AstNodeIndices.Children], result);
       result.value += '`';
-      return;
+      break;
     case '$p':
       serializeAst(node[AstNodeIndices.Children], result);
       result.value += '\n\n';
-      return;
+      break;
     case '$link':
       // The target is the first child of the link. We don't have to care if it's a placeholder
       // or not, because the serialization will automatically remove the extra empty.
-      const [target, ...children] = node[AstNodeIndices.Children];
+      const children = node[AstNodeIndices.Children], target = node[3]
```

Comment:
`Control` is actually the right one here! it's not super obvious, but here's the [definition context](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/crates/intl_markdown/src/icu/compile.rs#L69-L79), [type reference for consumers](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/packages/intl/src/types.d.ts#L195), and [usage in the formatter](https://github.com/discord/discord-intl/blob/1cab2d38e0581e166e4709de4958307902c04edc/packages/intl/src/format.ts#L148-L158).

tldr; Links are `Tag` elements, and tags have the additional `control` field to hold non-visible information, in this case the link target.

```suggestion
      const children = node[AstNodeIndices.Children];
      const target = node[AstNodeIndices.Control];
```